Detectors/FIT/FV0/base/src/Geometry.cxx

```diff
@@ -7,117 +7,1183 @@
+{
+  initializeCellRingRadii();
+  initializeSectorTransformations();
+  initializeFiberVolumeRadii();
+  initializeFiberMedium();
+  initializeScrewAndRodRadii();
+  initializeScrewTypeMedium();
+  initializeRodTypeMedium();
+  initializeScrewAndRodPositionsAndDimensions();
+}
 
-    TGeoMedium* kMed = gGeoManager->GetMedium("FV0_Scintillator$");
-    TGeoVolume* volCell = new TGeoVolume("FV0cell", geoCell, kMed);
+void Geometry::initializeCellRingRadii()
+{
+  // Index of mRAvgRing is NOT linked directly to any ring number
+  mRAvgRing.assign(sCellRingRadii, sCellRingRadii + sNumberOfCellRings + 1);
 
-    volCell->SetLineColor(kYellow);
-    sector->AddNode(volCell, iCell + 1);
+  // Set real scintillator radii (reduced by paint thickness and separation gap)
+  for (int i = 0; i < mRAvgRing.size() - 1; ++i) {
+    mRMinScintillator.push_back(mRAvgRing[i] + sDrSeparationScint);
+    mRMaxScintillator.push_back(mRAvgRing[i + 1] - sDrSeparationScint);
   }
-  return sector;
+  // Now indices of mRMinScint and mRMaxScint correspond to the same ring
+}
+
+void Geometry::initializeSectorTransformations()
+{
+  for (int iSector = 0; iSector < sNumberOfCellSectors; ++iSector) {
+    // iSector = 0 corresponds to the first sector clockwise from the y-axis
+    // iSector = 1 corresponds to the next sector in clockwise direction and so on
+
+    TGeoRotation* trans = createAndRegisterRot(sDetectorName + sSectorName + std::to_string(iSector) + "TRANS");
+
+    if (iSector == 2 || iSector == 3) {
+      // "a" and "b" mirrors.
+      // The reference to "a" and "b" can be understood with the CAD drawings of the detector.
+      trans->ReflectY(true);
+    }
+
+    mSectorTrans.push_back(trans);
+  }
+}
+
+void Geometry::initializeFiberVolumeRadii()
+{
+  mRMinFiber.push_back(0);
+  mRMinFiber.push_back(sDrMinContainerCone + sEpsilon);
+  mRMinFiber.push_back(sDrMinContainerFront + sEpsilon);
+
+  mRMaxFiber.push_back(sDrMinContainerCone);
+  mRMaxFiber.push_back(sDrMinContainerFront);
+  mRMaxFiber.push_back(mRMaxScintillator.back());
+}
+
+void Geometry::initializeFiberMedium()
+{
+  // There are no further checks if the medium is actually found
+  mMediumFiber.push_back(gGeoManager->GetMedium("FV0_FiberInner$"));
+  mMediumFiber.push_back(gGeoManager->GetMedium("FV0_FiberMiddle$"));
+  mMediumFiber.push_back(gGeoManager->GetMedium("FV0_FiberOuter$"));
+}
+
+void Geometry::initializeScrewAndRodRadii()
+{
+  mRScrewAndRod.push_back(mRAvgRing[1]);
+  mRScrewAndRod.push_back(mRAvgRing[2]);
+  mRScrewAndRod.push_back(mRAvgRing[3]);
+  mRScrewAndRod.push_back(mRAvgRing[4]);
+  mRScrewAndRod.push_back((mRAvgRing[4] + mRAvgRing[5]) / 2);
+  mRScrewAndRod.push_back(mRAvgRing[5]);
+}
+
+void Geometry::initializeScrewTypeMedium()
+{
+  // There are no further checks if the medium is actually found
+  TGeoMedium* medium = gGeoManager->GetMedium("FV0_Titanium$");
+  for (int i = 0; i < sNumberOfScrewTypes; ++i) {
+    mMediumScrewTypes.push_back(medium);
```

Comment:
are you planning to use in future different media for different screws? If not, what's is the point cashing vector of same medium pointers?
Same question for similar methods below.

Reply:
As long as the frame has not been manufactured and test-assembled, such small design details could change. And historically they did change. Initially the screws were supposed to be aluminum, but then the engineers started worrying about cold-welds in a long-term, so they switched to bronze or brass. In the most recent version they are made from titanium, but as such they are extremely expensive. 
That's why I would like to keep an option for easy change of screw and rod materials at least for the time being.